### DISCUSSION TITLE
Can I use a lower version of react and react-dom?

### DISCUSSION THREAD

**Discussion by jellli** [wakujs/waku#358] [Q&A] [answered]
I want to use some components I already have with Waku. But some of them can't work at react-dom@18.3.0, so is there any chance that I can use Waku in a lower version of react and react-dom?

**Comment by dai-shi** [comment, marked as answer]
Unfortunately, Waku, being RSC framework, requires the canary version of React.
Do you know why your component doesn't work with the canary version?

**Reply by jellli** [reply]
Thanks for your reply! I don't know the reason yet. It's a long story. The components are from an old project of my team's work, and they work in a specific version of React. I can not just upgrade them.

**Reply by dai-shi** [reply]
So, it's not a package, but just source code? If it's old source code, maybe it depends on some deprecated features.

**Reply by jellli** [reply]
Yes, there is nothing I can do about it. Thanks again!
